File: token.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/base64"
	"encoding/binary"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"time"
)

func main() {
	encodedkey := "RmFzdGx5IFRva2VuIFRlc3Q="
	var interval uint64 = 60

	key, err := base64.StdEncoding.DecodeString(encodedkey)
	if err != nil {
		log.Fatalf("error base64 decoding: %s", err)
	}

	number := Int64bytes(uint64(time.Now().Unix()) / interval)
	h := hmac.New(sha256.New, key)
	h.Write(number)

	token := base64.StdEncoding.EncodeToString(h.Sum(nil))

	response := FetchURL("http://token.fastly.com/token")
	validation := FetchURL("http://token.fastly.com?" + token)

	fmt.Println("Your Token:   ", token)
	fmt.Println("Fastly Token: ", string(response))
	fmt.Println("Validation:   ", string(validation))
}

func Int64bytes(int uint64) []byte {
	bytes := make([]byte, 8)
	binary.LittleEndian.PutUint64(bytes, uint64(int))
	return bytes
}

func FetchURL(url string) []byte {
	resp, err := http.Get(url)
	if err != nil {
		log.Fatalf("error fetching url %s : %s", url, err)
	}
	if resp.StatusCode != 200 {
		log.Fatalf("error fetching url %s : %s", url, resp.Status)
	}
	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		log.Fatalf("error reading body of http response: %s", err)
	}
	return body
}

File: token.php
<?

function pack64($value) {
  $l = ($value & 0xffffffff00000000) >>32;
  $r = $value & 0x00000000ffffffff;

  return pack('VV', $r, $l); 
}

function curl_get_contents($url)
{
  $ch = curl_init();
  curl_setopt($ch, CURLOPT_URL, $url);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
  curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
  $data = curl_exec($ch);
  curl_close($ch);
  return $data;
}

$encodedkey = "RmFzdGx5IFRva2VuIFRlc3Q=";
$interval = 60;
 
$key = base64_decode($encodedkey);
 
$number = pack64(intval(time()/$interval));
$token  = base64_encode(hash_hmac('sha256', $number, $key, true));

$response   = curl_get_contents("http://token.fastly.com/token");
$validation = curl_get_contents("http://token.fastly.com?$token");

print "Your Token:   $token\n";
print "Fastly Token: $response\n";
print "Validation:   $validation\n";

?>